Chat/assignment-chat/server-chat/src/main/java/ru/assignment/model/ChatModel.java

```diff
@@ -0,0 +1,60 @@
+package ru.assignment.model;
+
+import ru.assignment.message.ChatMessage;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Created by Андрей on 18.02.2015.
+ */
+public class ChatModel {
+    private Set<ChatModelListener> listenerSet;
+    private List<ChatMessage> messageList;
+
+    public ChatModel() {
+        listenerSet = new HashSet<ChatModelListener>();
+        messageList = new ArrayList<ChatMessage>();
+    }
+
+    public void addMessage(ChatMessage chatMessage,
+                           ChatModelListener chatModelListener) {
+        //messageSizeList++;
+        System.out.println("add messagetoChatModel");
+        messageList.add(chatMessage);
+        sendMessageToAllListeners(chatMessage, chatModelListener);
+    }
+
+    public void addListener(ChatModelListener listener) {
+        System.out.println("ChatModel add new listener");
+        listenerSet.add(listener);
+        System.out.println("model set sum "+listenerSet.size());
+    }
+
+    public void removeListener(int identifier) {
```

Comment:
removeListener(ChatModelListener listener)

using int will be ALWAYS wrong as Set object doesn't have remove by index, and even if it does, it's not a correct index, it's a session identificator.